test/plugins/webhooks.test.js

```diff
@@ -163,2314 +134,109 @@ describe('webhooks plugin test', () => {
-
-                pipelineMock.workflowGraph = releaseWorkflowMock;
-                pipelineMock.jobs = Promise.resolve([mainJobMock]);
-                pipelineFactoryMock.list.resolves([pipelineMock]);
-
-                return server.inject(options).then(reply => {
-                    assert.equal(reply.statusCode, 204);
-                    assert.notCalled(eventFactoryMock.create);
-                });
-            });
-
-            it('returns 201 with release non target branch pipeline release trigger', () => {
-                const releaseWorkflowMock = {
-                    nodes: [{ name: '~release' }, { name: 'main' }],
-                    edges: [{ src: '~release', dest: 'main' }]
-                };
-
-                parsed.branch = 'branch';
-                mainJobMock.requires = '~release';
-                pipelineMock.workflowGraph = releaseWorkflowMock;
-                pipelineMock.baxterthehacker = 'master';
-                pipelineMock.jobs = Promise.resolve([mainJobMock]);
-                pipelineFactoryMock.list.resolves([pipelineMock]);
-                pipelineFactoryMock.scm.parseUrl.resolves(scmUri);
-
-                return server.inject(options).then(reply => {
-                    assert.equal(reply.statusCode, 201);
-                    assert.calledWith(eventFactoryMock.create, {
-                        pipelineId: pipelineMock.id,
-                        type: 'pipeline',
-                        webhooks: true,
-                        username,
-                        scmContext,
-                        sha,
-                        configPipelineSha: latestSha,
-                        startFrom: '~release',
-                        baseBranch: 'branch',
-                        causeMessage: `Merged by ${username}`,
-                        changedFiles: undefined,
-                        releaseName: 'release01',
-                        ref: 'v0.0.1',
-                        meta: {
-                            sd: {
-                                release: {
-                                    id: 123456,
-                                    name: 'release01',
-                                    author: 'testuser'
-                                },
-                                tag: {
-                                    name: 'v0.0.1'
-                                }
-                            }
-                        }
-                    });
-                });
-            });
-
-            it('returns 204 when getCommitRefSha() is rejected', () => {
-                pipelineFactoryMock.scm.getCommitRefSha.rejects(new Error('some error'));
-
-                return server.inject(options).then(reply => {
-                    assert.equal(reply.statusCode, 204);
-                    assert.notCalled(eventFactoryMock.create);
-                });
+            return server.inject(options).then(() => {
+                assert.calledOnce(pipelineFactoryMock.scm.parseHook);
```

Comment:
Is this assertion necessary?

Reply:
There are too many test cases that check if parsedhook is called, so I will remove some of them. However, I want to check it somewhere, so I will keep one of them.